### ISSUE TITLE
Networked singleplayer

### ISSUE THREAD

**Issue by SimonN** [SimonN/LixD#334]
Lix 0.9.17.

It's possible to start networked games with only one team. Framestepping is disabled here.

**Ideas:**
* Many same-team players and many observes can start a networked singleplayer game, as they can now.
* Players may framestep, restart, pause, fast-forward, ..., as they can in offline singleplayer. These non-physics commands are transmitted and others try to re-compute to the desired frame, and continue at the desired speed (paused, normal, fast-forward, ...)
* Pause should be transferred with two different packets: pause-on and back-to-regular-speed, to avoid races over the network.
* Observers should *not* be able to framestep, restart, pause, ...
* Clicks on empty space should send a special packet that makes others highlight that location on the map. Or maybe transfer the entire mouse cursor location at all times?

**Comment by stavrossk** [comment]
Networked single player? What does it mean? Does it mean that you are playing alone but other people can spectate your game?

**Comment by SimonN** [comment]
Yes, that's the idea: One playes, and others spectate. Sometimes, I'm in voicechat with players or level designers, and it would be really helpful to spectate during singleplayer.

Networked games already allow several players to pick the same color, for a team game. It's natural, then, to allow several players to control the singleplayer color during networked singleplayer. Strictly speaking, it's not singleplayer puzzle-solving anymore, but rather single-team puzzle solving.
